ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1171,6 +1181,26 @@ public InstallSnapshotReplyProto installSnapshot(InstallSnapshotRequestProto req
     return reply;
   }
 
+  @Override
+  public PauseReplyProto pause(PauseRequestProto request) throws IOException {
+    assertLifeCycleState(LifeCycle.States.RUNNING);
+    // TODO: should pause() be limited on only working for a follower?
+    if (!lifeCycle.compareAndTransition(RUNNING, PAUSING)) {
+      return PauseReplyProto.newBuilder().setSuccess(false).build();
```

Comment:
Do we need the assert check here as lifecycle.compareAndTransition would check anyway before changing. 
It comes down to whether we want to throw an error if current state is not running or return a false as response.

Reply:
I see. I removed the assert check.